src/encoded/types/cart.py

```diff
@@ -87,12 +90,46 @@ def get_or_create_cart_by_user(context, request):
     user = request.registry[COLLECTIONS]['user'].get(userid)
     if not user:
         raise HTTPBadRequest()
-    carts = _get_carts_by_user(request, userid)
-    if not carts:
-        cart = _create_cart(request, user)
+    cart_name = None
+    all_carts = get_cart_objects_by_user(request, userid)
+    countable_carts = [cart for cart in all_carts if cart['status'] != 'deleted' and cart['status'] != 'disabled']
+    cart_atids = [cart['@id'] for cart in all_carts] if all_carts else []
+    cart_status = None
+    cart_name = None
+    try:
```

Comment:
I'm not sure we need these try/catches.  Can we try initializing the cart_status and cart_name vars directly from the request.json.get.  Default the .get('name', '') with the empty string to the strip won't fail.  If the keys don't exist both should be falsey.

Reply:
The first GET request to @@get-cart passes no JSON, so right now I broke into PDB on this line and get:

```
(Pdb) request
<Request at 0x1241bfe10 GET http://localhost:6543/carts/@@get-cart>
(Pdb) request.json
*** simplejson.scanner.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
(Pdb)
```

And if I let it go, I get a 400 error on the GET request. I’ll try @keenangraham’s suggestion of breaking out the GET and PUT to work around this, so that the PUT just expects JSON and the GET never does. I assume that would answer Keenan’s question just below this one as well.